webapp/tests/forms/steps/test_eligibility_steps.py

```diff
@@ -928,17 +966,42 @@ def test_if_post_and_session_data_correct_and_input_data_correct_than_set_next_i
         self.assertEqual(expected_url, step.render_info.next_url)
 
     def test_if_post_and_session_data_correct_and_input_data_incorrect_than_set_next_url_to_alternative_step(self):
-        session_data = {'marital_status_eligibility': 'divorced',
-                        'joint_taxes_eligibility': 'no'}
         with app.app_context() and app.test_request_context(method='POST', data={'alimony_eligibility': 'yes'}) as req:
-            req.session = SecureCookieSession({_ELIGIBILITY_DATA_KEY: create_session_form_data(session_data)})
+            req.session = SecureCookieSession({_ELIGIBILITY_DATA_KEY: create_session_form_data(self.correct_session_data)})
             step = EligibilityStepChooser('eligibility').get_correct_step(
                 SingleAlimonyDecisionEligibilityInputFormSteuerlotseStep.name)
             expected_url = step.url_for_step(SingleAlimonyEligibilityFailureDisplaySteuerlotseStep.name)
             step.handle()
 
         self.assertEqual(expected_url, step.render_info.next_url)
 
+    def test_if_session_data_correct_then_set_prev_input_step_correctly(self):
+        with app.app_context() and app.test_request_context(method='GET') as req:
+            req.session = SecureCookieSession({_ELIGIBILITY_DATA_KEY: create_session_form_data(self.correct_session_data)})
+            step = EligibilityStepChooser('eligibility').get_correct_step(
+                SingleAlimonyDecisionEligibilityInputFormSteuerlotseStep.name)
+            expected_url = step.url_for_step(DivorcedJointTaxesDecisionEligibilityInputFormSteuerlotseStep.name)
+            step.handle()
+        self.assertEqual(expected_url, step.render_info.prev_url)
+
+        alternative_data = {'marital_status_eligibility': 'single'}
```

Comment:
Maybe we could name these `alternative_data` variables more explicit (e.g. single_status_alternative_data in this case)?

Reply:
I am against that as it's quite some work, we will probably forget to rename the variable in the future, or in some cases even now and imo you can see the path within the dict. Sooooo, pretty strong no. 🙈

Reply:
In that case I would be more strongly for splitting the tests :)